src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -40,20 +50,33 @@ public void execute(X x) {
     PrintWriter         out            = x.get(PrintWriter.class);
     HttpServletResponse resp           = x.get(HttpServletResponse.class);
     HttpParameters      p              = x.get(HttpParameters.class);
-    CharBuffer          buffer_        = CharBuffer.allocate(65535);
-    String              serviceName    = p.getParameter("service");
-    String              methodName     = p.getParameter("method");
-    String              interfaceName  = p.getParameter("interfaceName");
+    String              data           = p.getParameter("data");
 
     try {
+      if ( SafetyUtil.isEmpty(data) ) {
```

Comment:
Is this an error if you call a method with no arguments?

Reply:
data contain the service, method, interfaceName and set of parameters. It should not be empty.